specs/protocol/derivation.md

```diff
@@ -1278,3 +1285,29 @@ follows:
 
 [extended-attributes]: exec-engine.md#extended-payloadattributesv1
 [Fee Vaults]: exec-engine.md#fee-vaults
+
+## On Future-Proof Transaction Log Derivation
+
+As described in [L1 Retrieval][#l1-retrieval], batcher transactions' types are required to be from a fixed allow-list.
+
+However, we want to allow deposit transactions and `SystemConfig` update events to get derived even from receipts of
+future transaction types, as long as the receipts can be decoded following a best-effort process:
```

Comment:
Lifting from this issue https://github.com/ethereum-optimism/optimism/issues/13627, should we:

> Clarify 7702 receipt type can contain valid deposits and system-config events

I'm not sure I understand this well myself at this point. An example might be good.